src/api/ResourcefulApi.js

```diff
@@ -21,25 +19,59 @@ export class ResourcefulApi extends Api {
    * @param {String} url
    * @param {Object} params
    * @param {Object} data
-   * @param {Object} options
    */
-  async doRequest (method, url, params, data, options) {
+  async _doRequest (method, url, params, data) {
     if (data) {
       data = this.preprocessData(data)
     }
 
-    return super.doRequest(method, url, params, data, options)
-      .then((response) => {
-        return {
-          data: normalize(response.data),
-          meta: response.data.meta,
-          status: response.status
+    return super._doRequest(method, url, params, data)
+      .then(async (response) => {
+        let json = {}
+
+        try {
+          json = await response.json()
+        } catch (e) {
+          if (response.status === 404) {
+            throw new NotFoundApiError('Resource not found')
+          }
+
+          throw new ApiError('Failed to decode response json')
         }
+
+        return this._parseResponse(response.status, json)
       })
   }
 
+  _parseResponse (status, json) {
```

Comment:
There is a lot more checking that should be done in this method which I purposefully left out because the current state already does more post processing than what happened in the axios days.